lace/test_obj.py

```diff
@@ -188,6 +188,49 @@ def test_reading_obj_with_mtl_from_s3_url(self, mock_s3_open):
         ])
         self.assertEqual(m.materials_filepath, self.obj_with_texure_mtl)
         self.assertEqual(m.texture_filepath, self.obj_with_texure_tex)
+        self.assertIsNotNone(m.texture_image)
+
+    def test_changing_texture_filepath(self):
+        m = obj.load(self.obj_with_texure)
+        self.assertEqual(m.texture_filepath, self.obj_with_texure_tex)
+        self.assertIsNotNone(m.texture_image)
+        m.texture_filepath = None
+        self.assertIsNone(m.texture_image)
+
+    def test_texture_in_Kd_vs_Ka(self):
```

Comment:
So a good unit test tests a small unit of code. As importantly, a good unit test as a statement. The name of the test should say something that makes sense to succeed or fail. This test _does_ test what you need covered, but it really is three separate tests merged into one. I'd go with something like the following:

```py
    def create_texture_test_files(self, include_Ka=False, include_Kd=False):
        obj_path = os.path.join(self.scratch_dir, 'texture_test.obj')
        with open(obj_path, 'w') as f:
            f.write('mtllib {}\n'.format('texture_test.mtl'))
        mtl_path = os.path.join(self.scratch_dir, 'texture_test.mtl')
        with open(mtl_path, 'w') as f:
            if include_Ka:
                f.write('map_Ka {}\n'.format('ambient_tex.png'))
            if include_Kd:
                f.write('map_Kd {}\n'.format('diffuse_tex.png'))
        return obj_path

    def test_texture_reads_Ka(self):
        obj_path = self.create_texture_test_files(include_Ka=True)
        m = obj.load(obj_path)
        self.assertEqual(m.texture_filepath, os.path.join(self.scratch_dir, 'ambient_tex.png'))

    def test_texture_reads_Kd(self):
        obj_path = self.create_texture_test_files(include_Kd=True)
        m = obj.load(obj_path)
        self.assertEqual(m.texture_filepath, os.path.join(self.scratch_dir, 'diffuse_tex.png'))

    def test_texture_reads_Ka_if_both_Ka_and_Kd_are_present(self):
        obj_path = self.create_texture_test_files(include_Ka=True, include_Kd=True)
        m = obj.load(obj_path)
        self.assertEqual(m.texture_filepath, os.path.join(self.scratch_dir, 'ambient_tex.png'))
```

That way when you get an error, it's much more immediately obvious what's failing, and if one of those components fails, the other two tests still run. Makes debugging a regression less painful. It also makes the test a much clearer assertion of what the behavior should look like.